src/App.tsx

```diff
@@ -1,51 +1,98 @@
-import React from 'react';
+import { Component } from 'react';
+import classNames from 'classnames';
 import './App.scss';
+import './Goods.scss';
 
-// import goodsFromServer from './goods';
-
-export const App: React.FC = () => {
-  return (
-    <main className="App">
-      <header className="App__header">
-        <h1 className="App__title">
-          Jam is selected
-        </h1>
-
-        <button
-          type="button"
-          className="App__clear"
-        >
-          Clear
-        </button>
-      </header>
-
-      <ul>
-        <li className="Good">Dumplings</li>
-        <li className="Good">Carrot</li>
-        <li className="Good">Eggs</li>
-        <li className="Good">Ice cream</li>
-        <li className="Good">Apple</li>
-        <li className="Good">Bread</li>
-        <li className="Good">Fish</li>
-        <li className="Good">Honey</li>
-        <li className="Good Good--active">Jam</li>
-        <li className="Good">Garlic</li>
-      </ul>
-
-      {/* Put required buttons into each Good */}
-      <button
-        type="button"
-        className="Good__remove"
-      >
-        Remove
-      </button>
-
-      <button
-        type="button"
-        className="Good__select"
-      >
-        Select
-      </button>
-    </main>
-  );
+import goodsFromServer from './goods';
+
+type State = {
+  selectedGoods: string[];
 };
+
+export class App extends Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  clearSelected = () => this.setState({ selectedGoods: [] });
+
+  removeGood = (good: string) => {
+    this.setState((prevState) => (
+      { selectedGoods: prevState.selectedGoods.filter(el => el !== good) }
+    ));
+  };
+
+  addGood = (good: string) => {
+    this.setState((prevState) => (
+      { selectedGoods: [...prevState.selectedGoods, good] }
+    ));
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+    const selectedCount = selectedGoods.length;
+    let selectedInfo = 'No goods selected';
+
+    if (selectedCount === 1) {
+      selectedInfo = `${selectedGoods} is selected`;
+    }
+
+    if (selectedCount > 1) {
+      selectedInfo = `${selectedGoods.slice(0, -1).join(', ')} and ${selectedGoods[selectedCount - 1]} are selected`;
+    }
+
+    return (
+      <main className="App panel is-primary">
+        <header className="App__header panel-heading">
+          <h1 className="App__title">
+            {selectedInfo}
+          </h1>
+
+          {selectedGoods.length && (
+            <button
+              type="button"
+              className="App__clear button is-danger"
+              onClick={this.clearSelected}
+            >
+              Clear
+            </button>
+          )}
+        </header>
+
+        <ul>
+          {goodsFromServer.map(good => (
+            <li
+              key={good}
+              className={classNames(
+                'Good',
+                { 'Good--active': selectedGoods.includes(good) },
```

Comment:
You are using the expression `selectedGoods.includes(good)` two time, let's move it to the variable.